### ISSUE TITLE
Destructuring into an empty object vs an object with existing properties yields different results

### ISSUE THREAD

**Issue by rivatove** [microsoft/TypeScript#60386]
### 🔎 Search Terms

"destructuring", "record"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.6.3#code/MYewdgzgLgBATgU1HAJgdQJZQBYGUpwZgDmEAXDAEpIioA80hJANDI0cQHwwC8MA3gF8A3AChRCAB4AHWrFCRYiZOizYAcgFcAtgCMEcclRr12LGGB3643Pv1EwYUggEMACnBDSDUAJ4UABmYHGAA6cOVaVRx8JlJgkSA

### 💻 Code

This raises a type error:

```ts
const recordWithStrings: Record<string, string> = {};

export const recordWithNumbers: Record<string, number> = {
  ...recordWithStrings,
};
```

and this doesn't:

```ts
const recordWithStrings: Record<string, string> = {};

export const recordWithNumbers: Record<string, number> = {
  extraProperty: 0,
  ...recordWithStrings,
};
```

Is there a sound reason for that?

### 🙁 Actual behavior

The 2nd example doesn't raise a type error.

### 🙂 Expected behavior

The 2nd example should raise a type error.

**Comment by jcalz** [comment]
Maybe related to #27144?

**Comment by Andarist** [comment]
The reported code is almost an exact copy of what was reported in this comment https://github.com/microsoft/TypeScript/issues/27273#issuecomment-1114703301 and here https://github.com/microsoft/TypeScript/issues/56431 . The last experiment to fix this was worked on here https://github.com/microsoft/TypeScript/pull/56441

**Comment by zanminkian** [comment]
I am facing the same issue. This bug makes spread operator not type-safe.

**Comment by RyanCavanaugh** [comment]
This is the same index-signature spread issue tracked in [#56431](https://github.com/microsoft/TypeScript/issues/56431). In both cases, adding an explicit property to an object literal causes a spread `Record<string, V>` to escape checking against the target `Record<string, U>`, allowing values that do not satisfy the target index signature.
